### PR TITLE
Fix: remove @obj class and extension signatures

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Remove @objc signatures which cause errors when building XCFramework.

### CODE REVIEW

File: Source/UserSession/ZMClientRegistrationStatus.m [hunk 1]
```diff
@@ -305,39 +301,6 @@ - (void)fetchExistingSelfClientsAfterClientRegistered:(UserClient *)currentSelfC
     }
 }
 
-- (void)didFailToRegisterClient:(NSError *)error
```
Comment (typfel) [inline Source/UserSession/ZMClientRegistrationStatus.m:308]: Why was is necessary to convert this method to Swift?

**Review by typfel — COMMENTED** [review]
So the reason why you need to remove all `objc` annotation and extension is because  the classes inherits from `AbstractRequestStrategy` which is annotated with `@objcMembers`?

File: Source/UserSession/ZMClientRegistrationStatus.m (hunk 1 above)
Comment (billypchan) [reply Source/UserSession/ZMClientRegistrationStatus.m:308]: This charge is made here:
https://github.com/wireapp/wire-ios-sync-engine/pull/1447

I will change the base branch of the PR.

**Comment by billypchan** [comment]
@typfel yes, `AbstractRequestStrategy` subclassing is not possible if we do not update the iOS version to 13.0.

Wire 2021-06-02 at 11_58 AM

File: Source/Synchronization/Strategies/ConversationRoleDownstreamRequestStrategy.swift [hunk 2]
```diff
@@ -77,10 +76,7 @@ public final class ConversationRoleDownstreamRequestStrategy: AbstractRequestStr
     public var requestGenerators: [ZMRequestGenerator] {
         return [downstreamSync]
     }
-}
-
 
-extension ConversationRoleDownstreamRequestStrategy: ZMDownstreamTranscoder {
```
Comment (David-Henner) [inline Source/Synchronization/Strategies/ConversationRoleDownstreamRequestStrategy.swift:83]: Is there a specific reason for removing the extensions?

File: Source/Synchronization/Strategies/TeamDownloadRequestStrategy.swift [hunk 3]
```diff
@@ -91,7 +91,8 @@ fileprivate extension Team {
 
 /// Responsible for downloading the team which the self user belongs to during the slow sync
 /// and for updating it when processing events or when manually requested.
-public final class TeamDownloadRequestStrategy: AbstractRequestStrategy, ZMContextChangeTrackerSource {
+    
+public final class TeamDownloadRequestStrategy: AbstractRequestStrategy, ZMContextChangeTrackerSource, ZMEventConsumer, ZMSingleRequestTranscoder, ZMDownstreamTranscoder {
```
Comment (David-Henner) [inline Source/Synchronization/Strategies/TeamDownloadRequestStrategy.swift:95]: what's the reason for moving the methods from `TeamDownloadRequestStrategy+Events` to this file?

File: Source/Synchronization/Strategies/ConversationRoleDownstreamRequestStrategy.swift (hunk 2 above)
Comment (billypchan) [reply Source/Synchronization/Strategies/ConversationRoleDownstreamRequestStrategy.swift:83]: if we do not set iOS version to iOS 13, this line will give compile error. It is solved by moving the protocol extension to class definition

File: Source/Synchronization/Strategies/TeamDownloadRequestStrategy.swift (hunk 3 above)
Comment (billypchan) [reply Source/Synchronization/Strategies/TeamDownloadRequestStrategy.swift:95]: same as above.

**Review by David-Henner — APPROVED** [review]

**Review by typfel — APPROVED** [review]

**Pull request merged**